### ISSUE TITLE
Boolean group should not override all values

### ISSUE THREAD

**Issue by krystof-k** [avo-hq/avo#2603]
### Describe the bug / Steps to Reproduce

Consider a following `metadata` column:

```json
{
  "feature_enabled": true,
  "another_feature_enabled": false,
  "something_else": "some_value"
}
```

with a boolean group configured

```ruby
field :metadata,
  as: :boolean_group,
  options: { "feature_enabled": "Feature", "another_feature_enabled": "Another feature" }
```

When a resource with such field is saved, it overrides the whole `metadata` column:

```json
{
  "feature_enabled": true,
  "another_feature_enabled": true
}
```

### Expected behavior & Actual behavior

Only values listed in options should be changed.

### System configuration
**Avo version**: 3.5.3

**Rails version**: 6.1.7.7

**Ruby version**: 3.1.4

**License type**:

 - [x] Community
 - [ ] Pro
 - [ ] Advanced

**Are you using Avo monkey patches, overriding views or view components?**

 - [ ] Yes. If so, please post code samples.
 - [x] No

#### Impact

 - [ ] High impact (It makes my app un-usable.)
 - [x] Medium impact (I'm annoyed, but I'll live.)
 - [ ] Low impact (It's really a tiny thing that I could live with.)

#### Urgency

 - [ ] High urgency (I can't continue development without it.)
 - [x] Medium urgency (I found a workaround, but I'd love to have it fixed.)
 - [ ] Low urgency (It can wait. I just wanted you to know about it.)

**Comment by adrianthedev** [comment]
Hey @krystof-k. Thanks for writing about this.

We haven't thought about that use case, and we are not completely sure we want to support it.
It would be more logic than we wished for behind the scenes which will make it more error prone.

Try to add this monkeypatch and customize the `fill_field` method to suit your needs.
I'd love to see the final form and see if it's a fit for upstreaming to Avo.

```ruby
module Avo
  module Fields
    class KeyValueField < BaseField
      def fill_field(model, key, value, params)
        # change logic here
        begin
          new_value = JSON.parse(value)
        rescue
          new_value = {}
        end

        model.send("#{key}=", new_value)

        model
      end
    end
  end
end
```

**Comment by krystof-k** [comment]
Hey @adrianthedev, thanks, yeah, I'll try to come up with a PR. I don't think it would be so complicated, actually I think it is more dangerous when it behaves like this, someone can lose data.

**Comment by adrianthedev** [comment]
I agree that a warning is required there.
I added it https://docs.avohq.io/3.0/fields/key_value.html

**Comment by krystof-k** [comment]
Cool :) But I'm talking boolean group the whole time, not key value, which is however the same case I assume.

**Comment by adrianthedev** [comment]
You're right. Something didn't make sense.
Thank you!

I moved the warning.

https://docs.avohq.io/3.0/fields/boolean_group.html

**Comment by krystof-k** [comment]
Hey @adrianthedev, I fixed it (#2677), it is rather a minor change, I'd love to see it merged. Regarding the key value field, as there is not possible to specify the options, this issue doesn't occur in that case.
